File: src/dataset/cpp/05-log-levels_exercism-ARPIT73881/openAI/ChatGPT4_log_levels_v4.cpp
#include <string>

namespace log_line {
    std::string message(const std::string& msg) {
        return msg.substr(msg.find(' ') + 1);
    }
    std::string log_level(const std::string& msg) {
        auto end = msg.find(']');
        return msg.substr(1, end - 1);
    }
    std::string reformat(const std::string& msg) {
        auto start = msg.find(' ') + 1;
        auto end = msg.find(']');
        return msg.substr(start) + " (" + msg.substr(1, end - 1) + ")";
    }
}

File: src/dataset/Java/log-levels_exercism-shashwata27/openAI/ChatGPT4_LogLevels_v4.java
import java.util.Locale;

public class LogLevels {

    public static String message(String logLine) {
        return logLine.substring(logLine.indexOf(':') + 1).strip();
    }

    public static String logLevel(String logLine) {
        int start = logLine.indexOf('[') + 1;
        int end = logLine.indexOf(']');
        return logLine.substring(start, end).toLowerCase(Locale.ROOT);
    }

    public static String reformat(String logLine) {
        int colonIndex = logLine.indexOf(':');
        int start = logLine.indexOf('[') + 1;
        int end = logLine.indexOf(']');
        return logLine.substring(colonIndex + 1).strip() + " (" + logLine.substring(start, end).toLowerCase(Locale.ROOT) + ")";
    }
}